src/utils/price_comparison_utils.ts

```diff
@@ -53,139 +54,148 @@ const emptyPlaceholderSources = Object.values(ERC20BridgeSource).reduce<SourceCo
-                      const aGasCostTakerAssetUnitAmount = a.gas
-                          .times(quote.gasPrice)
-                          .dividedBy(ethUnitAmount)
-                          .times(quote.sellTokenToEthRate);
-                      const bGasCostTakerAssetUnitAmount = b.gas
-                          .times(quote.gasPrice)
-                          .dividedBy(ethUnitAmount)
-                          .times(quote.sellTokenToEthRate);
-
-                      const aTotal = a.unitTakerAmount.plus(aGasCostTakerAssetUnitAmount);
-                      const bTotal = b.unitTakerAmount.plus(bGasCostTakerAssetUnitAmount);
-
-                      return aTotal.comparedTo(bTotal);
-                  });
-            // Select the best (first in the sorted list) option for each source
-            const uniqueSources = _.uniqBy(sortedSources, 'liquiditySource');
-
-            const sourcePrices: SourceComparison[] = uniqueSources.map(source => {
-                const { liquiditySource, unitMakerAmount, unitTakerAmount, gas } = source;
-
-                // NOTE: In order to not communicate a price better than the actual quote we
-                // should make sure to always round towards a worse price
-                const roundingStrategy = isSelling ? BigNumber.ROUND_FLOOR : BigNumber.ROUND_CEIL;
-
-                const price = isSelling
-                    ? unitMakerAmount.dividedBy(unitTakerAmount).decimalPlaces(buyToken.decimals, roundingStrategy)
-                    : unitTakerAmount.dividedBy(unitMakerAmount).decimalPlaces(sellToken.decimals, roundingStrategy);
-
-                return {
-                    name: liquiditySource,
-                    price,
-                    gas,
-                };
-            });
-
-            // Add null values for all sources we don't have a result for so that we always have a full result set in the response
-            const allSourcePrices = _.uniqBy<SourceComparison>([...sourcePrices, ...emptyPlaceholderSources], 'name');
-
-            return allSourcePrices;
-        } catch (err) {
-            logger.error(err, `Failed to calculate price comparisons`);
+            return getPriceComparisonFromQuoteOrThrow(chainId, side, quote);
+        } catch (e) {
+            logger.error(`Error calculating price comparisons, skipping [${e}]`);
+            return undefined;
         }
     },
+    renameNative(sc: SourceComparison): SourceComparison {
+        return {
+            ...sc,
+            name: sc.name === ERC20BridgeSource.Native ? '0x' : sc.name,
+        };
+    },
 };
+
+interface PartialQuote {
+    buyTokenAddress: string;
+    sellTokenAddress: string;
+    buyAmount: BigNumber;
+    sellAmount: BigNumber;
+    sellTokenToEthRate: BigNumber;
+    buyTokenToEthRate: BigNumber;
+    gasPrice: BigNumber;
+    estimatedGas: BigNumber;
+    quoteReport?: QuoteReport;
+}
+
+function getPriceComparisonFromQuoteOrThrow(
```

Comment:
NIT: This fn never intentionally throws so the naming here might be a bit misleading. I originally put it inside a try/catch to guard against unexpected run-time errors in the calculations erroring out the entire quote request. It should never (🙏 ) error out under normal circumstances.